File: packages/datapaw-context/src/semantic_config/models/metric_lib.py
from __future__ import annotations

from pydantic import BaseModel


class MetricCreate(BaseModel):
    datasource_id: str | None = None  # 数据源对外编码
    domain_id: int | None = None
    metric_name: str | None = None
    description: str | None = None
    unit: str | None = None
    is_polaris: bool | None = False
    show_distribution: bool | None = False
    is_visible: bool | None = True
    synonyms: str | None = None
    tags: str | None = None


class MetricUpdate(BaseModel):
    metric_name: str | None = None
    description: str | None = None
    unit: str | None = None
    is_polaris: bool | None = None
    show_distribution: bool | None = None
    is_visible: bool | None = None
    synonyms: str | None = None
    tags: str | None = None


class MetricResponse(BaseModel):
    id: int
    datasource_id: str | None = None  # 数据源对外编码
    domain_id: int | None = None
    datasource_name: str | None = None
    domain_name: str | None = None
    metric_name: str | None = None
    description: str | None = None
    unit: str | None = None
    is_polaris: bool | None = None
    show_distribution: bool | None = None
    is_visible: bool | None = None
    synonyms: str | None = None
    tags: str | None = None

File: packages/qwenpaw-data-context/frontend/src/types/metricLib.ts

// 指标库数据
export interface MetricLibItem {
  // 主键
  id?: number;
  // 指标名称
  metric_name: string;
  // 数据源 ID
  datasource_id?: string;
  // 业务域 ID
  domain_id?: number;
  // 数据源名称（展示用）
  datasource_name?: string;
  // 业务域名称（展示用）
  domain_name?: string;
  // 描述
  description?: string;
  // 单位
  unit?: string;
  // 是否北极星指标
  is_polaris?: boolean;
  // 是否展示分布
  show_distribution?: boolean;
  // 是否可见
  is_visible?: boolean;
  // 同义词
  synonyms?: string;
  // 标签
  tags?: string;
  // 数据集ID（兼容旧字段）
  dataset_id?: string;
  // 数据集名称
  dataset_name?: string;
  // 域（兼容旧文本字段，逐步废弃）
  domain?: string;
}

// 查询指标库参数
export interface QueryMetricLibParams {
  metric_name?: string;
  datasource_id?: string;
  domain_id?: number;
  synonyms?: string;
  page?: number;
  size?: number;
}

// 指标重复检查结果
export interface MetricDuplicate {
  existing_metric: string;
  similarity_judgment: string;
  reason: string;
  suggestion: string;
}

export interface MetricCheckResult {
  input_metric: string;
  domain?: string;
  suggestion: string;
}
